builder/comp-builder.nix

```diff
@@ -294,13 +294,23 @@ stdenv.mkDerivation ({
   '';
 
   # Note: Cabal does *not* copy test executables during the `install` phase.
+  #
+  # Note 2: if a package contains multiple libs (lib + sublibs) SETUP register will generate a
+  #         folder isntead of a file for the configuration.  Therfore if the result is a folder,
+  #         we need to register each and every element of that folder.
```

Comment:
Shouldn't there only be one package per derivation? Why would there be multiple?

Reply:
I think the issue is the `all` target.  If you run the following on a cabal file with a sublibrary:

```
$ runghc Setup.hs configure # without component target, as we do for `all`.
$ runghc Setup.hs register --gen-pkg-config=pkg.conf
```
you end up with a folder for `pkg.conf` that contains the package configuration file for the library as well as the sublibrary.

If you don't have sublibraries, `pkg.conf` is your package configuration file.

Reply:
@angerman So this only comes up when a package depends on the `all` component? That shouldn't happen, should it?

Reply:
Why should this not happen? Why wouldn't we build the `all` component? We do allow this right now (and the test case tried to do precisely that).

Thus, unless we want to remove the `all` component again, I'm afraid we'll need to keep this.

Reply:
@angerman The all component is, to my understanding, only for the sake of users who want a shell for all, plus the novel purpose of building all components in one derivation. The intention is that components will depend on components, not on any `all` derivation.